### PR TITLE
Feature/bc 462 sync chat with figma

### PR DESCRIPTION
**Pull request by jasonmar08**

### CODE REVIEW

**Review by HectorIlarraza — CHANGES_REQUESTED** [review]
Overall a great job Jason! you have evolved and stepped up ur coding game to a new level, I learned so much reviewing ur PR. There are some minor areas that need to be changed and suggestions I made for your code that I hope makes it more fluid and easier to read. Please let me know if they are helpful after giving them some thought.

**Review by HectorIlarraza — COMMENTED** [review]
Terrific work Jason just some questions that would help me understand if the methods suggested helped in any way. Once you answer them ill approve ASAP!

File: src/components/ChatDialog/ChatDialogMain/ChatDialogMain.tsx [hunk 1]
```diff
@@ -43,70 +99,119 @@ export const ChatDialogMain = () => {
...
+const ChatTitle = ({
+  chatScreen,
+  onBackArrowClick,
+  onScreenUpdate,
+  currentConversation,
+  selectedMember,
+  profilePictures,
+  getTitleText,
+}) => {
+  const titleContentLookup = {
```
[the remark this answers is not in the document]
Comment (jasonmar08) [reply src/components/ChatDialog/ChatDialogMain/ChatDialogMain.tsx:119]: Yeah, first time I used a lookup table and I like it. If the occasion allows for it, I think it's a great option and it avoids the sequential evaluation that if-else statements and switch statements encompass.

File: src/interfaces/ChatInterface.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+export interface ChatSliceInterface {
```
[the remark this answers is not in the document]
Comment (jasonmar08) [reply src/interfaces/ChatInterface.ts:1]: I already had the interface set up in the userSlice and transferred it to a chatSlice. I think it separates the logic for chat from authUser, making the code more organized and maintanable.

File: src/utils/data/chatConstants.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {
```
[the remark this answers is not in the document]
Comment (jasonmar08) [reply src/utils/data/chatConstants.ts:1]: Yes, the constants helped! I like how organized they are now, and it also makes it easier if another developer were to work on the chat. If you type `ChatScreen.`, a list of all the chatScreens appears so you can select your desired one.

File: src/utils/redux/slices/chatSlice.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit'
```
[the remark this answers is not in the document]
Comment (jasonmar08) [reply src/utils/redux/slices/chatSlice.ts:1]: I think separating the chatSlice makes it more organized and maintainable.

**Review by HectorIlarraza — APPROVED** [review]

**Pull request merged**